app/celery/process_comp_and_pen.py

```diff
@@ -0,0 +1,119 @@
+from dataclasses import dataclass
+from uuid import uuid4
+
+from flask import current_app
+from sqlalchemy.orm.exc import NoResultFound
+
+from notifications_utils.statsd_decorators import statsd
+
+from app import notify_celery
+from app.dao.service_sms_sender_dao import dao_get_service_sms_sender_by_id
+from app.models import (
+    Service,
+    Template,
+)
+from app.notifications.send_notifications import lookup_notification_sms_setup_data, send_notification_bypass_route
+from app.va.identifier import IdentifierType
+
+
+@dataclass
+class DynamoRecord:
+    participant_id: str
+    payment_amount: str
+    vaprofile_id: str
+
+
+@notify_celery.task(name='comp-and-pen-batch-process')
+@statsd(namespace='tasks')
+def comp_and_pen_batch_process(records: list[dict[str, str]]) -> None:
+    """Process batches of Comp and Pen notification requests.
+
+    Args:
+        records (list[dict[str, str]]): The incoming records
+    """
+    current_app.logger.debug(f'comp_and_pen_batch_process records: {records}')
+
+    # Grab all the necessary data
+    try:
+        service, template, sms_sender_id = lookup_notification_sms_setup_data(
+            current_app.config['COMP_AND_PEN_SERVICE_ID'],
+            current_app.config['COMP_AND_PEN_TEMPLATE_ID'],
+            current_app.config['COMP_AND_PEN_SMS_SENDER_ID'],
+        )
+        reply_to_text = dao_get_service_sms_sender_by_id(service.id, sms_sender_id).sms_sender
+    except (AttributeError, NoResultFound, ValueError):
+        current_app.logger.exception('Unable to send comp and pen notifications due to improper configuration')
+        raise
+
+    _send_comp_and_pen_sms(
+        service,
+        template,
+        sms_sender_id,
+        reply_to_text,
+        [DynamoRecord(**item) for item in records],
```

Comment:
nitty and no fix required. 

Double iteration - We're iterating over all the records to convert them in a DynamoRecord object. Then in ` _send_comp_and_pen_sms` we are iterating over thru all the records again. Could we do both these things in the same iteration?

Reply:
That's a good callout. I had gone back and forth on creating the dataclass at each iteration vs in a single call. I opted to go this route for simplicity in the downstream method since the other methods acts as a "pre-processor".